#! /usr/bin/env python

PACKAGE='youbot_scanner_lines'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *


gen = ParameterGenerator()
#       Name                    					Type      Reconfiguration level             Description                                                                                      Default    Min   Max
gen.add("scanner_lines_clusterMinSize",				int_t,	   0,								"The Minimum size of a cluster.",																5,			5,    200000)
gen.add("scanner_lines_clusterMaxSize",				int_t,	   0,								"The Maximum size of a cluster.",																25000,		5,    200000)
gen.add("scanner_lines_clusterTolerance",			double_t,  0,								"The tolerance in meters for the EuclideanCluster filter, whether to include a point in the cluster", 0.1,  0.001, 100.0)
gen.add("scanner_lines_clusterCloud",				bool_t,	   0,								"Whether to cluster before processing the lines or not.")
gen.add("scanner_lines_keepCloudFraction",   	    double_t,  0, 							    "The double-fraction of the cloud which is left at maximum, when the processing has finished.",             0.1,    	0.1, 	  1.0)
gen.add("scanner_lines_cropMinX",    				double_t,  0,    							"The minimum x coordinate part in meters of the box to crop out of the cloud before processing lines.",            0,     -100.0,    100.0)
gen.add("scanner_lines_cropMinY",    				double_t,  0,    							"The minimum y coordinate part in meters of the box to crop out of the cloud before processing lines.",            -0.1,     -100.0,    100.0)
gen.add("scanner_lines_cropMaxX",    				double_t,  0,    							"The maximum x coordinate part in meters of the box to crop out of the cloud before processing lines.",            0.5,     -100.0,    100.0)
gen.add("scanner_lines_cropMaxY",    				double_t,  0,    							"The maximum y coordinate part in meters of the box to crop out of the cloud before processing lines.",            0.1,     -100.0,    100.0)
gen.add("scanner_lines_sacModelDistance",    		double_t,  0,    							"The distance threshold of the SAC model.",                                    0.01,     0.001,        2.0)
exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "ScannerLines"))
